### DISCUSSION TITLE
Can Not Find The File"libdiscon.dll"

### DISCUSSION THREAD

**Discussion by Tang960** [NatLabRockies/ROSCO#256] [Q&A]
Hello,
  When I run the example code "04_simple_sim.py" from the ROSCO Toolbox, it is error because of the lack of file "libdiscon.dll". Where can I find this file?
  Thanks and regards

**Comment by dzalkind** [comment]
Hi, you can install it using the instructions on readthedocs or download it from the releases page.

**Comment by yusufarikan** [comment]
I can't find the libdiscon.dll file either, can you please give a specific link. Is the libdiscon.dll a generic file or has to be re-generated every time *.IN file is updated?

Thanks

**Reply by dzalkind** [reply]
https://rosco.readthedocs.io/en/latest/source/install.html#installing-the-rosco-tools 

The libdiscon dynamic library does not need to be recompiled every time an *.IN file is generated.

**Comment by t0ng-duck** [comment]
Hi,
  I want to learn the source code for libdiscon.dll. How can I get it？

**Reply by dzalkind** [reply]
Here is the source code: https://github.com/NREL/ROSCO/tree/main/ROSCO/src

Here are instructions for compiling it: https://rosco.readthedocs.io/en/latest/source/install.html
